### PR TITLE
Using junit5 with auto detect

### PR DESCRIPTION
**Pull request by vrdhn**

### CODE REVIEW

**Comment by bremac** [comment]
Using the ConsoleLauncher looks good to me, but we need to combine `JAVA_TEST_CLASSES` and `JAVA_ALL_TEST_CLASSES` into the test JAR so that the launcher discovers them. See https://github.com/sightmachine/rocksdb/pull/4/files#diff-31ed790e942fadafe334a13765d2379b7b89b1c3d65a3973a94d55a77624c5db

File: .circleci/config.yml [hunk 1]
```diff
@@ -158,6 +158,8 @@ jobs:
             mkdir test-jars/
             cp java/test-libs/*.jar test-jars/
             cp build/java/rocksdbjni_test_classes.jar test-jars/
+            rm test-jars/*junit-4*
```
Comment (bremac) [inline .circleci/config.yml:161]: I think we could just remove the JUnit 4 JAR from the CMakeLists.txt?

Comment (vrdhn) [reply .circleci/config.yml:161]: well, these guys have their own s3 bucket from where they download java deps in the CMake.
The Right way is obviously to put junit5 in there, ... so i just chose the [worst](https://www.jwz.org/doc/worse-is-better.html) way (-:

**Review by bremac — APPROVED** [review]
LGTM. I'm going to get all of the test classes into the test JAR and then merge this. :+1:

**Pull request merged**